libass/ass_outline.c

```diff
@@ -209,13 +220,34 @@ bool outline_scale_pow2(ASS_Outline *outline, const ASS_Outline *source,
         return true;
     }
 
+    int32_t lim_x = OUTLINE_MAX;
+    if (scale_ord_x > 0)
+        lim_x = scale_ord_x < 32 ? lim_x >> scale_ord_x : 0;
+    else
+        scale_ord_x = FFMAX(scale_ord_x, -32);
+
+    int32_t lim_y = OUTLINE_MAX;
+    if (scale_ord_y > 0)
+        lim_y = scale_ord_y < 32 ? lim_y >> scale_ord_y : 0;
+    else
+        scale_ord_y = FFMAX(scale_ord_y, -32);
```

Comment:
Doesn’t this (if it ever is a no-op) affect the numbers and cause the outline to be scaled out of proportion? Shouldn’t this clamping be done on `k->scale_ord_x` as soon as it is obtained in `get_bitmap_glyph`?

Reply:
Negative scale is an equivalent to the arithmetic shift right and such a shift of `int32_t` value reduces to the sign bit for shift amounts from 31 and up. So `-32` here can even be replaced with `-31` without any influence on the result.